### PR TITLE
gracefully handle artifacts' API unavailability

### PR DESCRIPTION
**Pull request by pkasprzyk**
ref. https://neptune-labs.atlassian.net/browse/NPT-10843

### CODE REVIEW

**Review by HubertJaworski — APPROVED** [review]

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -165,10 +170,14 @@ def __init__(self, credentials: Credentials, proxies: Optional[Dict[str, str]] =
...
-        self.artifacts_client = create_swagger_client(
-            build_operation_url(self._client_config.api_url, self.ARTIFACTS_SWAGGER_PATH),
-            self._http_client
-        )
+        try:
+            self.artifacts_client = create_swagger_client(
+                build_operation_url(self._client_config.api_url, self.ARTIFACTS_SWAGGER_PATH),
+                self._http_client
+            )
+        except Exception:
```
Comment (aniezurawski) [inline neptune/new/internal/backends/hosted_neptune_backend.py:178]: A lot of different things can go wrong here. Exception does not necessarily mean artifacts are not supported.
Maybe we should add list of supported features to config returned from backend.
@Herudaio how urgent is it to make new clients work for onprems?

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 2]
```diff
@@ -184,11 +193,18 @@ def __init__(self, credentials: Credentials, proxies: Optional[Dict[str, str]] =
             python_version=platform.python_version())
         self._http_client.session.headers.update({'User-Agent': user_agent})
 
+        if self.missing_features:
+            warnings.warn(NeptuneFeaturesNotAvailableWarning(self.missing_features).message)
```
Comment (aniezurawski) [inline neptune/new/internal/backends/hosted_neptune_backend.py:197]: I am not a fan of printing this in every single Run.

**Review by aniezurawski — CHANGES_REQUESTED** [review]

File: neptune/new/internal/backends/hosted_neptune_backend.py (hunk 2 above)
Comment (pkasprzyk) [reply neptune/new/internal/backends/hosted_neptune_backend.py]: it's (now) a python warning, so the use can setup it to print only once
`warnings.simplefilter("once", NeptuneFeaturesNotAvailableWarning)`
or choose to ignore it alltogether
`warnings.simplefilter("ignore", NeptuneFeaturesNotAvailableWarning)`

either way, your fight is with spec, not me ;)

Comment (pkasprzyk) [reply neptune/new/internal/backends/hosted_neptune_backend.py]: spec has changed, so removing

**Review by aniezurawski — APPROVED** [review]

**Pull request merged**